aten/src/ATen/native/native_functions.yaml

```diff
@@ -4526,26 +4526,14 @@
     CPU: channel_shuffle_cpu
     CompositeImplicitAutograd: math_channel_shuffle
 
-- func: is_pinned(Tensor self, Device? device=None) -> bool
+- func: is_pinned(Tensor self) -> bool
   variants: method
-  dispatch:
-    NestedTensorCUDA, CUDA: is_pinned_cuda
-    MPS: is_pinned_mps
-    CompositeExplicitAutograd: is_pinned_default
 
 # TODO: add a copy kwarg that guarantees that the tensor is put into fresh
 # pinned memory
-- func: pin_memory(Tensor(a) self, Device? device=None) -> Tensor(a)
+- func: pin_memory(Tensor(a) self) -> Tensor(a)
```

Comment:
This function's implementation is NOT Composite. So we have to add a CompositeExplicitAutograd: pin_memory here.
This will lead to the fact that you must add back the derivatives formula and the aliasing information must be accurate.

Since this function sometimes return a view and sometimes not, you won't be able to have the aliasing info correctly and will have to do the same trick as before:
pin_memory remain compositeimplicit (no specific dispatch for it) and can have inncurate aliasing info as before.
You have a _pin_memory that is never aliasing that is compositeexplicitautograd that is accurate and has a derivatives.yaml formula.

tl;dr: you have to restore the _pin_memory trick I'm afraid to have proper autograd/aliasing behavior.
Also since you have to keep that, you might want to keep the _pin_memory_nested as well on top of the compositeexplicitimplementation

Reply:
Yes, you're right. I ignored the aliasing information before. And it's difficult to use CompositeExplicitAutograd: pin_memory here.
So I have restored the _pin_memory trick and use CompositeImplicitAutograd: pin_memory to  guarantee proper autograd behavior.
Also for _pin_memory, keep the _pin_memory_nested on top of the CompositeExplicitAutograd. Here, I only reserve NestedTensorCPU key, cause manual backendselect for _pin_memory is removed and input tensor `self` can be performed pin_memory operation only if it's a cpu tensor. So unlike before, NestedTensorCUDA key and other nested keys are just meaningless for _pin_memory now. When inputting a nested cuda tensor, the behavior changes from throwing a RuntimeError before("only dense CPU tensors can be pinned") to throwing a NotImplementedError now ("Could not run aten::_pin_memory with arguments from the NestedTensorCUDA backend").